app/models/manageiq/providers/embedded_ansible/automation_manager/job.rb

```diff
@@ -3,8 +3,91 @@ class ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Job < ManageIQ::P
 
   require_nested :Status
 
+  def self.run_playbook(options, userid = nil)
+    userid ||= 'system'
+    name = build_name(options[:name])
+    description = options[:description]
+    info = options[:config_info]
+    playbook = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook.find(info[:playbook_id])
+    tower = playbook.manager
+    info[:inventory] = create_and_return_inventory(tower, name, description) if info[:inventory].nil?
+    _tower, params = ServiceTemplateAnsiblePlaybook.build_parameter_list(name, description, info)
+    job = nil
+    tower.with_provider_connection do |connection|
+      # Launch shouldn't need options since extra_vars can be included in the creation
+      #   of the job template
+      job = connection.api.job_templates.create!(params).launch
+      job.to_hash
+    end
+    requeue_job_status(userid, options, job)
+  end
+
+  def self.reload_job_status(job, options)
+    info = options[:config_info]
+    playbook = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook.find(info[:playbook_id])
+    tower = playbook.manager
+    tower.with_provider_connection do |connection|
+      job = connection.api.jobs.find(job['id'])
+      job.to_hash
+    end
+  end
+
+  def self.requeue_job_status(userid, options, job)
+    job = reload_job_status(job, options)
+    if job['status'] == 'successful' || job['status'] == 'failed'
+      job['status']
+    else
+      userid ||= 'system'
```

Comment:
I can probably move all the queuing code into its own method since the two calls are largely the same.